src/test-harness/SparkLiquidityLayerTests.sol

```diff
@@ -346,50 +565,48 @@ abstract contract SparkLiquidityLayerTests is SpellRunner {
 
         SparkLiquidityLayerContext memory ctx = _getSparkLiquidityLayerContext();
 
-        // NOTE: Using param here because during an upgrade the _sparkLiquidityLayerContext
-        //       will return a controller that is out of date.
-        ForeignController domainController = ForeignController(foreignController);
-
         address domainAlmProxy = address(ctx.proxy);
 
         uint256 domainUsdcProxyBalance = domainUsdc.balanceOf(domainAlmProxy);
         uint256 domainUsdcPsmBalance   = domainUsdc.balanceOf(domainPsm3);
 
         assertEq(domainUsdc.balanceOf(domainAlmProxy), domainUsdcProxyBalance);
 
-        _relayMessageOverBridges();
+        //_relayMessageOverBridges();
+        CCTPBridgeTesting.relayMessagesToDestination(bridge, true);
 
         assertEq(domainUsdc.balanceOf(domainAlmProxy), domainUsdcProxyBalance + usdcAmount);
         assertEq(domainUsdc.balanceOf(domainPsm3),     domainUsdcPsmBalance);
 
         // --- Step 3: Deposit USDC into PSM3 ---
 
         vm.prank(ctx.relayer);
-        domainController.depositPSM(address(domainUsdc), usdcAmount);
+        foreignController.depositPSM(address(domainUsdc), usdcAmount);
 
         assertEq(domainUsdc.balanceOf(domainAlmProxy), domainUsdcProxyBalance);
         assertEq(domainUsdc.balanceOf(domainPsm3),     domainUsdcPsmBalance + usdcAmount);
 
         // --- Step 4: Withdraw all assets from PSM3 ---
 
         vm.prank(ctx.relayer);
-        domainController.withdrawPSM(address(domainUsdc), usdcAmount);
+        foreignController.withdrawPSM(address(domainUsdc), usdcAmount);
 
         assertEq(domainUsdc.balanceOf(domainAlmProxy), domainUsdcProxyBalance + usdcAmount);
         assertEq(domainUsdc.balanceOf(domainPsm3),     domainUsdcPsmBalance);
 
         // --- Step 5: Bridge USDC back to mainnet ---
 
         vm.prank(ctx.relayer);
-        domainController.transferUSDCToCCTP(usdcAmount, CCTPForwarder.DOMAIN_ID_CIRCLE_ETHEREUM);
+        foreignController.transferUSDCToCCTP(usdcAmount, CCTPForwarder.DOMAIN_ID_CIRCLE_ETHEREUM);
 
         assertEq(domainUsdc.balanceOf(domainAlmProxy), domainUsdcProxyBalance);
 
         chainSpellMetadata[ChainIdUtils.Ethereum()].domain.selectFork();
 
         assertEq(usdc.balanceOf(Ethereum.ALM_PROXY), mainnetUsdcProxyBalance);
 
-        _relayMessageOverBridges();
+        //_relayMessageOverBridges();
```

Comment:
Rm comment

Reply:
This is to fix the issue with the fork/storage. I added a FIXME comment.